qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -13,380 +13,85 @@
-
-        # recursive step on half the angles fulfilling the above assumption
-        multiplex_1 = self._multiplex(target_gate, list_of_angles[0 : (list_len // 2)], False)
-        circuit.append(multiplex_1.to_instruction(), q[0:-1])
-
-        # attach CNOT as follows, thereby flipping the LSB qubit
-        circuit.append(CXGate(), [msb, lsb])
-
-        # implement extra efficiency from the paper of cancelling adjacent
-        # CNOTs (by leaving out last CNOT and reversing (NOT inverting) the
-        # second lower-level multiplex)
-        multiplex_2 = self._multiplex(target_gate, list_of_angles[(list_len // 2) :], False)
-        if list_len > 1:
-            circuit.append(multiplex_2.to_instruction().reverse_ops(), q[0:-1])
-        else:
-            circuit.append(multiplex_2.to_instruction(), q[0:-1])
-
-        # attach a final CNOT
-        if last_cnot:
-            circuit.append(CXGate(), [msb, lsb])
-
-        return circuit
-
-    def broadcast_arguments(self, qargs, cargs):
-        flat_qargs = [qarg for sublist in qargs for qarg in sublist]
-
-        if self.num_qubits != len(flat_qargs):
-            raise QiskitError(
-                "Initialize parameter vector has %d elements, therefore expects %s "
-                "qubits. However, %s were provided."
-                % (2**self.num_qubits, self.num_qubits, len(flat_qargs))
-            )
-        yield flat_qargs, []
-
-    def validate_parameter(self, parameter):
-        """Initialize instruction parameter can be str, int, float, and complex."""
-
-        # Initialize instruction parameter can be str
-        if self._from_label:
-            if parameter in ["0", "1", "+", "-", "l", "r"]:
-                return parameter
-            raise CircuitError(
-                "invalid param label {} for instruction {}. Label should be "
-                "0, 1, +, -, l, or r ".format(type(parameter), self.name)
-            )
-
-        # Initialize instruction parameter can be int, float, and complex.
-        if isinstance(parameter, (int, float, complex)):
-            return complex(parameter)
-        elif isinstance(parameter, np.number):
-            return complex(parameter.item())
-        else:
-            raise CircuitError(f"invalid param type {type(parameter)} for instruction  {self.name}")
+        return self._stateprep.gates_to_uncompute()
 
 
 def initialize(self, params, qubits=None):
-    r"""Initialize qubits in a specific state.
+    """Initialize qubits in a specific state.
 
     Qubit initialization is done by first resetting the qubits to :math:`|0\rangle`
-    followed by an state preparing unitary. Both these steps are included in the
-    `Initialize` instruction.
+    followed by calling :class:`~.qiskit.extensions.quantum_initializer.StatePreparation`
+    class to prepare the qubits in a specified state.
+    Both these steps are included in the
+    :class:`~.qiskit.extensions.quantum_initializer.Initialize` instruction.
```

Comment:
I might be wrong, Jake is the expert on this, but I don't think the leading `.` is needed/should be here
```suggestion
    followed by calling :class:`~qiskit.extensions.quantum_initializer.StatePreparation`
    class to prepare the qubits in a specified state.
    Both these steps are included in the
    :class:`~qiskit.extensions.quantum_initializer.Initialize` instruction.
```

Reply:
Julien's right - it's not necessary (but also won't really hurt) if you're using the full search path.  Here I suspect that the Sphinx object is actually called `qiskit.extensions.Initialize` - the path isn't based on the full Python specification, but which docstring the autosummary link is in (technically the value of Sphinx' `currentmodule`).  The `.` is useful when you want to do a partial suffix match, so you can do things like `` :class:`.StatePreparation` `` and it'll search the directory for the object you mean and raise an error if there's multiple matches.